test/e2e/capabilityargocd_test.go

```diff
@@ -110,6 +112,62 @@ func assertArgoCDCreated(ctx context.Context, t *testing.T, cfg *envconf.Config)
 	return ctx
 }
 
+func assertArgoCDUpdated(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+	updatedRevision := "updatedRevision"
+	paas := getPaas(ctx, paasWithArgo, t, cfg)
+	paas.Spec.Capabilities = api.PaasCapabilities{
+		"argocd": api.PaasCapability{
+			Enabled:          true,
+			SshSecrets:       map[string]string{paasArgoGitUrl: paasArgoSecret},
+			GitUrl:           paasArgoGitUrl,
+			GitPath:          paasArgoGitPath,
+			GitRevision:      updatedRevision,
+			ExtraPermissions: true,
+		},
+	}
+
+	if err := updatePaasSync(ctx, cfg, paas); err != nil {
+		t.Fatal(err)
+	}
+
+	// As no reconciliation of a PaasNs is triggered, we sleep for safety
+	time.Sleep(10 * time.Second)
```

Comment:
This is a long sleep; if `updatePaasSync` already waits for Paas reconciliation to complete, doesn't that mean the Argo application has already been updated? Or is there another race condition I'm not considering?